src/Security/MemberAuthenticator/LostPasswordHandler.php

```diff
@@ -197,16 +189,55 @@ public function forgotPassword($data, $form)
 
         // Avoid information disclosure by displaying the same status,
         // regardless wether the email address actually exists
+        $identifier = isset($data[$uniqueIdentifier]) ? $data[$uniqueIdentifier] : '';
         $link = Controller::join_links(
             $this->link('passwordsent'),
-            rawurlencode($data['Email']),
+            rawurlencode($identifier),
             '/'
         );
 
         return $this->redirect($this->addBackURLParam($link));
     }
 
     /**
+     * Ensure that the user has provided an email address. Note that the "Email" key is specific to this
+     * implementation, but child classes can override this method to use another unique identifier field
+     * for validation.
+     *
+     * @param  array $data
+     * @param  LostPasswordForm $form
+     * @return HTTPResponse|null
+     */
+    protected function validateForgotPasswordData(array $data, LostPasswordForm $form)
+    {
+        if (empty($data['Email'])) {
+            $form->sessionMessage(
+                _t(
+                    'SilverStripe\\Security\\Member.ENTEREMAIL',
+                    'Please enter an email address to get a password reset link.'
+                ),
+                'bad'
+            );
+
+            return $this->redirectToLostPassword();
+        }
+    }
+
+    /**
+     * Load an existing Member from the provided data
+     *
+     * @param  array $data
+     * @param  string $uniqueIdentifier
+     * @return Member|null
+     */
+    protected function getMemberFromData(array $data, $uniqueIdentifier)
+    {
+        if (!empty($data[$uniqueIdentifier])) {
+            return Member::get()->filter([$uniqueIdentifier => $data[$uniqueIdentifier]])->first();
```

Comment:
This is the $data[] array key access I was referring to.

Reply:
Ok I've switch this back to `$data['Email']` as it was originally